lib/i18n.py

```diff
@@ -30,7 +30,9 @@
 from typing import Dict
 from collections import namedtuple
 
-LOCALE_DIR = os.path.join(os.path.dirname(__file__), 'locale')
+from . import locale  # Causes translation files to be extracted on Android.
+
+LOCALE_DIR = os.path.dirname(locale.__file__)
```

Comment:
Don't you mean `locale.__path__` here? Serious question.  `__file__` is not always defined is it?

Reply:
Oh now it is because you added `__init__.py`. Hmm. Ok..

Reply:
`__file__` should be defined for all modules, except built-in ones like `sys`.

Reply:
It is only defined if the module has an `__init__.py`. See 00bbe37051e4c5e2be5db385ee59a624d6b92c3f.

We had to also add this as a package to `setup.py` now as well so that Windows, Mac, and Linux binary builds don't crash.

This could have been avoided had  my genius clever suggestion of `__path__` been used. :P

Reply:
Oh right, sorry about that. I guess it must have been treating it as an implcit namespace package (PEP 420), which doesn't have a `__file__` either.